src/content/en/fundamentals/engage-and-retain/push-notifications/subscribing-a-user.md

```diff
@@ -0,0 +1,396 @@
+do that but a tiny tip is to use `JSON.strinigify()` to get all the necessary data out of the
+subscription object, otherwise you can piece together the same result manually:
+
+    const subscriptionObject = {
+      endpoint: pushSubscription.endpoint,
+      keys: {
+        p256dh: pushSubscription.getKeys('p256dh'),
+        auth: pushSubscription.getKeys('auth')
+      }
+    };
+
+    // The above is the same output as:
+
+    const subscriptionObjectToo = JSON.stringify(pushSubscription);
+
+In [the demo used throughout this
+book](https://github.com/gauntface/web-push-book/tree/master/src/demos/node-server), we make a
+POST request to send a subscription to our node server that stores the subscription in a
+database.
+
+Sending the subscription is done in the web page like so:
+
+    function sendSubscriptionToBackEnd(subscription) {
+      return fetch('/api/save-subscription/', {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(subscription)
+      })
+      .then(function(response) {
+        if (!response.ok) {
+          throw new Error('Bad status code from server.');
+        }
+
+        return response.json();
+      })
+      .then(function(responseData) {
+        if (!(responseData.data && responseData.data.success)) {
+          throw new Error('Bad response from server.');
+        }
+      });
+    }
+
+The node server receives this request and saves the data to a database for use later on.
+
+    app.post('/api/save-subscription/', function (req, res) {
+      if (!isValidSaveRequest(req, res)) {
+        return;
+      }
+
+      return saveSubscriptionToDatabase(req.body)
+      .then(function(subscriptionId) {
+        res.setHeader('Content-Type', 'application/json');
+        res.send(JSON.stringify({ data: { success: true } }));
+      })
+      .catch(function(err) {
+        res.status(500);
+        res.setHeader('Content-Type', 'application/json');
+        res.send(JSON.stringify({
+          error: {
+            id: 'unable-to-save-subscription',
+            message: 'The subscription was received but we were unable to save it to our database.'
+          }
+        }));
+      });
+    });
+
+With the `PushSubscription` details on our server we are good to send our user
+a message whenever we want.
+
+## FAQs
+
+A few common questions people have asked at this point:
+
+> Can I change the push service a browser uses?
+
+No. The push service is selected by the browser and as we saw with the
+`subscribe()` call, the browser will make network requests to the push service
+to retrieve the details that make up the *PushSubscription*.
+
+> Each browser uses a different Push Service, don't they have different API's?
+
+All push services will expect the same API.
+
+This common API is called the `Web Push Protocol` and describes the network request / API call
+your application will need to make to trigger a push message.
+
```

Comment:
Think about explaining this near the top. If you do, this question may recede over time.